<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Function reference â€¢ Rubrary</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">Rubrary</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.13.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/DE_Genes.html">Differential Gene Expression</a>
    <a class="dropdown-item" href="../articles/GSEA.html">Gene Set Enrichment Analysis (GSEA)</a>
    <a class="dropdown-item" href="../articles/PCA_Quickstart.html">Principal Component Analysis (PCA) Quickstart</a>
    <a class="dropdown-item" href="../articles/PCA_Walkthrough.html">Principal Component Analysis (PCA) Walkthrough</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/liaoyjruby/Rubrary/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Function reference</h1>
    </div>

    <div class="section level2">
      <h2 id="exploratory-data-analysis">Exploratory data analysis<a class="anchor" aria-label="anchor" href="#exploratory-data-analysis"></a></h2>
      
      

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="plot_distribution.html">plot_distribution()</a></code> 
        </dt>
        <dd>Plot distribution of numeric vector</dd>
      </dl><dl><dt>
          
          <code><a href="plot_density.html">plot_density()</a></code> 
        </dt>
        <dd>Density plot by group</dd>
      </dl><dl><dt>
          
          <code><a href="plot_scatter.html">plot_scatter()</a></code> 
        </dt>
        <dd>Plot simple scatter</dd>
      </dl><dl><dt>
          
          <code><a href="get_kspval.html">get_kspval()</a></code> 
        </dt>
        <dd>Calculate (signed) Kolgomorov-Smirnov enrichment p-value in dataframe</dd>
      </dl><dl><dt>
          
          <code><a href="check_normal.html">check_normal()</a></code> 
        </dt>
        <dd>Shapiro-Wilk test of normality</dd>
      </dl><dl><dt>
          
          <code><a href="calc_sumZscore.html">calc_sumZscore()</a></code> 
        </dt>
        <dd>Generate Sum Z Score</dd>
      </dl><dl><dt>
          
          <code><a href="run_RRHO.html">run_RRHO()</a></code> 
        </dt>
        <dd>Run RRHO analysis</dd>
      </dl><dl><dt>
          
          <code><a href="phyper_df.html">phyper_df()</a></code> 
        </dt>
        <dd>Hypergeometric test, 0 vs. non-0, on columns of a dataframe</dd>
      </dl></div><div class="section level2">
      <h2 id="principal-components-analysis-pca-">Principal components analysis (PCA)<a class="anchor" aria-label="anchor" href="#principal-components-analysis-pca-"></a></h2>
      
      

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="run_PCA.html">run_PCA()</a></code> 
        </dt>
        <dd>Run PCA (<code>prcomp</code> wrapper)</dd>
      </dl><dl><dt>
          
          <code><a href="plot_PCA.html">plot_PCA()</a></code> 
        </dt>
        <dd>Plot PCA with annotation</dd>
      </dl><dl><dt>
          
          <code><a href="plot_PCA_matrix.html">plot_PCA_matrix()</a></code> 
        </dt>
        <dd>Plot PCA matrix via <code><a href="https://ggobi.github.io/ggally/reference/ggpairs.html" class="external-link">GGally::ggpairs</a></code></dd>
      </dl><dl><dt>
          
          <code><a href="plot_PCA_3D.html">plot_PCA_3D()</a></code> 
        </dt>
        <dd>Plot 3D PCA scores/loadings via <code>plotly</code></dd>
      </dl><dl><dt>
          
          <code><a href="plot_PCA_biplot.html">plot_PCA_biplot()</a></code> 
        </dt>
        <dd>Plot PCA biplot</dd>
      </dl><dl><dt>
          
          <code><a href="get_loadings.html">get_loadings()</a></code> 
        </dt>
        <dd>Extract loadings from <code>prcomp</code> object</dd>
      </dl><dl><dt>
          
          <code><a href="plot_screeplot.html">plot_screeplot()</a></code> 
        </dt>
        <dd>Plot screeplot from prcomp</dd>
      </dl><dl><dt>
          
          <code><a href="rotate_varimax.html">rotate_varimax()</a></code> 
        </dt>
        <dd>Apply varimax rotation to PCA scores</dd>
      </dl><dl><dt>
          
          <code><a href="predict_PCA.html">predict_PCA()</a></code> 
        </dt>
        <dd>PCA prediction by projecting query/test samples onto PCA of reference/train samples</dd>
      </dl></div><div class="section level2">
      <h2 id="differentially-expressed-genes-deseq-">Differentially expressed genes (DESeq)<a class="anchor" aria-label="anchor" href="#differentially-expressed-genes-deseq-"></a></h2>
      
      

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="output_DESeq.html">output_DESeq()</a></code> 
        </dt>
        <dd>Output DESeq results as dataframe with signed log p metric</dd>
      </dl><dl><dt>
          
          <code><a href="plot_DEgene_boxplot.html">plot_DEgene_boxplot()</a></code> 
        </dt>
        <dd>Plot gene expression boxplots of selected genes (from DESeq signature)</dd>
      </dl><dl><dt>
          
          <code><a href="plot_waterfall.html">plot_waterfall()</a></code> 
        </dt>
        <dd>Plot horizontal waterfall plot (with optional density)</dd>
      </dl><dl><dt>
          
          <code><a href="plot_volcano.html">plot_volcano()</a></code> 
        </dt>
        <dd>Plot (DE genes) volcano plot</dd>
      </dl></div><div class="section level2">
      <h2 id="gene-set-enrichment-analysis-gsea-">Gene set enrichment analysis (GSEA)<a class="anchor" aria-label="anchor" href="#gene-set-enrichment-analysis-gsea-"></a></h2>
      
      <p class="section-desc"></p><p>Including GSEA Squared functions.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="format_GSEA_name.html">format_GSEA_name()</a></code> 
        </dt>
        <dd>Format GSEA pathway name</dd>
      </dl><dl><dt>
          
          <code><a href="plot_GSEA_barplot.html">plot_GSEA_barplot()</a></code> 
        </dt>
        <dd>Plot GSEA NES barplot</dd>
      </dl><dl><dt>
          
          <code><a href="plot_GSEA_pathway.html">plot_GSEA_pathway()</a></code> 
        </dt>
        <dd>Plot GSEA pathway plots (waterfall + enrichment)</dd>
      </dl><dl><dt>
          
          <code><a href="plot_GSEA_pathway_batch.html">plot_GSEA_pathway_batch()</a></code> 
        </dt>
        <dd><code>plot_GSEA_pathway</code> wrapper that works nicely with <code>lapply</code></dd>
      </dl><dl><dt>
          
          <code><a href="get_GSEAsq_terms.html">get_GSEAsq_terms()</a></code> 
        </dt>
        <dd>Get GSEA Squared terms/keywords</dd>
      </dl><dl><dt>
          
          <code><a href="run_GSEA_squared.html">run_GSEA_squared()</a></code> 
        </dt>
        <dd>Run GSEA Squared on GSEA results</dd>
      </dl><dl><dt>
          
          <code><a href="plot_GSEAsq_density.html">plot_GSEAsq_density()</a></code> 
        </dt>
        <dd>Plot categorical percentile rank comparison between two GSEA squared signatures</dd>
      </dl></div><div class="section level2">
      <h2 id="single-cell-rna-analysis">Single cell RNA analysis<a class="anchor" aria-label="anchor" href="#single-cell-rna-analysis"></a></h2>
      
      <p class="section-desc"></p><p>Seurat object based analysis and visualization.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="plot_dimplot_grid.html">plot_dimplot_grid()</a></code> 
        </dt>
        <dd>Plot Seurat DimPlot grid set</dd>
      </dl><dl><dt>
          
          <code><a href="plot_comp_barplot.html">plot_comp_barplot()</a></code> 
        </dt>
        <dd>Plot stacked barplot filled by group</dd>
      </dl><dl><dt>
          
          <code><a href="plot_featurescatter.html">plot_featurescatter()</a></code> 
        </dt>
        <dd>Plot FeatureScatter wrapper</dd>
      </dl><dl><dt>
          
          <code><a href="assess_integration.html">assess_integration()</a></code> <code><a href="assess_integration.html">run_LISI()</a></code> <code><a href="assess_integration.html">run_MixingMetric()</a></code> <code><a href="assess_integration.html">run_CellMixS()</a></code> 
        </dt>
        <dd>Compute and plot integration assessment cell-specific methods for Seurat object</dd>
      </dl><dl><dt>
          
          <code><a href="genecoexp_scatter_hyper.html">genecoexp_scatter_hyper()</a></code> 
        </dt>
        <dd>Plot pairwise gene coexpression scatter and calculate hypergeometric enrichment p-value between two genelists</dd>
      </dl></div><div class="section level2">
      <h2 id="gene-manipulation">Gene manipulation<a class="anchor" aria-label="anchor" href="#gene-manipulation"></a></h2>
      
      

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="convert_genes.html">convert_genes()</a></code> 
        </dt>
        <dd>Convert list of genes via BioMart</dd>
      </dl><dl><dt>
          
          <code><a href="get_PC_genes.html">get_PC_genes()</a></code> 
        </dt>
        <dd>Retrieve protein coding genes from Ensembl BioMart</dd>
      </dl><dl><dt>
          
          <code><a href="filter_genes.html">filter_genes()</a></code> 
        </dt>
        <dd>Filter dataframe by list of (protein coding) genes</dd>
      </dl><dl><dt>
          
          <code><a href="get_gene_desc.html">get_gene_desc()</a></code> 
        </dt>
        <dd>Get gene description from BioMart</dd>
      </dl><dl><dt>
          
          <code><a href="get_log_coding_geneexp.html">get_log_coding_geneexp()</a></code> 
        </dt>
        <dd>Get log(x+1) protein coding gene expression matrix</dd>
      </dl></div><div class="section level2">
      <h2 id="utility">Utility<a class="anchor" aria-label="anchor" href="#utility"></a></h2>
      
      <p class="section-desc"></p><p>Miscellaneous helper functions.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="rread.html">rread()</a></code> 
        </dt>
        <dd>Wrapper for <code><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">data.table::fread</a></code> implementing <code>row.names</code> functionality</dd>
      </dl><dl><dt>
          
          <code><a href="rwrite.html">rwrite()</a></code> 
        </dt>
        <dd>Wrapper for <code><a href="https://Rdatatable.gitlab.io/data.table/reference/fwrite.html" class="external-link">data.table::fwrite</a></code> with tab separated values (TSV) default parameters</dd>
      </dl><dl><dt>
          
          <code><a href="corner.html">corner()</a></code> 
        </dt>
        <dd>Like head but just the corner</dd>
      </dl><dl><dt>
          
          <code><a href="split_line.html">split_line()</a></code> 
        </dt>
        <dd>Split text into multiple lines</dd>
      </dl><dl><dt>
          
          <code><a href="full_join_rownames.html">full_join_rownames()</a></code> 
        </dt>
        <dd>Full join by rownames</dd>
      </dl><dl><dt>
          
          <code><a href="left_join_rownames.html">left_join_rownames()</a></code> 
        </dt>
        <dd>Left join by rownames</dd>
      </dl><dl><dt>
          
          <code><a href="cbind.fill.html">cbind(<i>&lt;fill&gt;</i>)</a></code> 
        </dt>
        <dd>Combine data.frames by column, filling in missing rows.</dd>
      </dl><dl><dt>
          
          <code><a href="use_pkg.html">use_pkg()</a></code> 
        </dt>
        <dd>Check if package is installed, prompt if not</dd>
      </dl><dl><dt>
          
          <code><a href="inst_pkg.html">inst_pkg()</a></code> 
        </dt>
        <dd>Prompt user to install package</dd>
      </dl><dl><dt>
          
          <code><a href="reload_pkg.html">reload_pkg()</a></code> 
        </dt>
        <dd>Reload package by installation path</dd>
      </dl></div><div class="section level2">
      <h2 id="data">Data<a class="anchor" aria-label="anchor" href="#data"></a></h2>
      
      <p class="section-desc"></p><p>Used for vignettes and examples.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="airway_deseq_res.html">airway_deseq_res</a></code> 
        </dt>
        <dd>Airway Dex Treated vs. Untreated DESeq2 Results</dd>
      </dl><dl><dt>
          
          <code><a href="GSEA_pathways.html">GSEA_pathways</a></code> 
        </dt>
        <dd>Pathways from <code>msigdbr</code> to use with <code>fgsea</code></dd>
      </dl></div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Yi Jou (Ruby) Liao.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

