<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Great for sanity checking, to make sure that differentially expressed genes as reported through DE gene analysis metrics (such as DESeq2) are reflected in gene expression data per sample."><title>Plot gene expression boxplots of selected genes (from DESeq signature) — plot_DEgene_boxplot • Rubrary</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Plot gene expression boxplots of selected genes (from DESeq signature) — plot_DEgene_boxplot"><meta property="og:description" content="Great for sanity checking, to make sure that differentially expressed genes as reported through DE gene analysis metrics (such as DESeq2) are reflected in gene expression data per sample."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">Rubrary</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.10.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/DE_Genes.html">Differential Gene Expression</a>
    <a class="dropdown-item" href="../articles/GSEA.html">Gene Set Enrichment Analysis (GSEA)</a>
    <a class="dropdown-item" href="../articles/PCA_Quickstart.html">Principal Component Analysis (PCA) Quickstart</a>
    <a class="dropdown-item" href="../articles/PCA_Walkthrough.html">Principal Component Analysis (PCA) Walkthrough</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/liaoyjruby/Rubrary/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Plot gene expression boxplots of selected genes (from DESeq signature)</h1>
      <small class="dont-index">Source: <a href="https://github.com/liaoyjruby/Rubrary/blob/HEAD/R/DEgenes.R" class="external-link"><code>R/DEgenes.R</code></a></small>
      <div class="d-none name"><code>plot_DEgene_boxplot.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Great for sanity checking, to make sure that differentially expressed genes as reported through DE gene analysis metrics (such as DESeq2) are reflected in gene expression data per sample.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">plot_DEgene_boxplot</span><span class="op">(</span></span>
<span>  <span class="va">data</span>,</span>
<span>  <span class="va">anno</span>,</span>
<span>  <span class="va">anno_sig</span>,</span>
<span>  anno_type <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">sig_DE</span>,</span>
<span>  genes <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  metric <span class="op">=</span> <span class="st">"sign_log_p"</span>,</span>
<span>  pval <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  genes_n <span class="op">=</span> <span class="fl">6</span>,</span>
<span>  colors <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  title <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  savename <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  height <span class="op">=</span> <span class="fl">8</span>,</span>
<span>  width <span class="op">=</span> <span class="fl">8</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>data</dt>
<dd><p>df/string; (path to) numeric gene expression matrix, rownames = genes, colnames = samples</p></dd>


<dt>anno</dt>
<dd><p>df/string; (path to) annotation table, rownames = samples, columns = metadata</p></dd>


<dt>anno_sig</dt>
<dd><p>string; colname in <code>anno</code> to be used as breaks for x-axis</p></dd>


<dt>anno_type</dt>
<dd><p>string; colname in <code>anno</code> to be used to color jitter points by</p></dd>


<dt>sig_DE</dt>
<dd><p>df/string, optional; (path to) DE(seq) gene results table</p></dd>


<dt>genes</dt>
<dd><p>char vector; list of genes to plot, must be present in <code>data</code></p></dd>


<dt>metric</dt>
<dd><p>string; colname in <code>sig_DE</code> to rank genes by for auto gene selection if <code>genes</code> not provided and for appending to gene name in parens</p></dd>


<dt>pval</dt>
<dd><p>logical; use <code>t.test</code> pvalue annotation via <code><a href="https://rpkgs.datanovia.com/ggpubr/reference/stat_compare_means.html" class="external-link">ggpubr::stat_compare_means</a></code></p></dd>


<dt>genes_n</dt>
<dd><p>integer; if <code>genes</code> not provided, get top/bottom <code>genes_n</code></p></dd>


<dt>colors</dt>
<dd><p>char vector; jitter point colors, length == # of unique <code>anno_type</code>s</p></dd>


<dt>title</dt>
<dd><p>string; plot title</p></dd>


<dt>savename</dt>
<dd><p>string; file path to save plot under</p></dd>


<dt>height</dt>
<dd><p>numeric; plot height</p></dd>


<dt>width</dt>
<dd><p>numeric; plot width</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>ggplot object, with facet-wrapped boxplot</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Providing DE(seq) gene signature (<code>sig_DE</code>) is optional; if not provided, <code>genes</code> list must be defined. If DE gene info <em>is</em> provided, each subplot's title will have the <code>metric</code> value appended in parenthesis.</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Yi Jou (Ruby) Liao.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

