<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Rubrary">
<title>DESeq / GSEA • Rubrary</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="DESeq / GSEA">
<meta property="og:description" content="Rubrary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">Rubrary</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.8.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/DESeq_GSEA.html">DESeq / GSEA</a>
    <a class="dropdown-item" href="../articles/PCA.html">Principal Components Analysis (PCA)</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/liaoyjruby/Rubrary/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>DESeq / GSEA</h1>
                        <h4 data-toc-skip class="author">Yi Jou (Ruby)
Liao</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/liaoyjruby/Rubrary/blob/HEAD/vignettes/DESeq_GSEA.Rmd" class="external-link"><code>vignettes/DESeq_GSEA.Rmd</code></a></small>
      <div class="d-none name"><code>DESeq_GSEA.Rmd</code></div>
    </div>

    
    
<p>Setup</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://liaoyjruby.github.io/Rubrary/">Rubrary</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># For `airway` install</span></span>
<span><span class="va">r</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"repos"</span><span class="op">)</span></span>
<span><span class="va">r</span><span class="op">[</span><span class="st">"CRAN"</span><span class="op">]</span> <span class="op">=</span> <span class="st">"http://cran.us.r-project.org"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>repos <span class="op">=</span> <span class="va">r</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The example dataset for this vignette comes from the paper <span class="citation">[1]</span> below:</p>
<blockquote>
<p>Himes et al. (2014). “RNA-Seq Transcriptome Profiling Identifies
CRISPLD2 as a Glucocorticoid Responsive Gene that Modulates Cytokine
Function in Airway Smooth Muscle Cells.” PLoS ONE, 9(6), e99625. <a href="https://doi.org/10.1371/journal.pone.0099625" class="external-link uri">https://doi.org/10.1371/journal.pone.0099625</a>.</p>
</blockquote>
<p>From the abstract:</p>
<blockquote>
<p>Using RNA-Seq, a high-throughput sequencing method, we characterized
transcriptomic changes in four primary human ASM cell lines that were
treated with dexamethasone—a potent synthetic glucocorticoid (1 µM for
18 hours).</p>
</blockquote>
<p>The experiment was performed on human airway smooth muscle (ASM)
cells. Glucocorticoids are a commonly used drug to treat asthma for its
anti-inflammatory effects in lung tissue. The goal of the experiment was
to characterize the mechanism of glucocorticoid inflammation suppression
through RNAseq. Himes et al. found a number of glucocorticoid-responsive
genes such as DUSP1 and CRISPLD2. They also identified enrichment in
functional annotation of terms such as “glycoprotein/extracellular
matrix”, “vasculature development”, and “circulatory system
process”.</p>
<p>In the original paper, the authors used <a href="https://www.nature.com/articles/nbt.2450" class="external-link">Cuffdiff 2</a> to find
differential gene expression and used the <a href="https://david.ncifcrf.gov/" class="external-link">NIH Database for Annotation,
Visualization, and Integrated Discovery (DAVID)</a> for gene set
enrichment analysis.</p>
</div>
<div class="section level2">
<h2 id="load-data">Load data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h2>
<p>The package <code>airway</code> contains data from the Himes et
al. paper in the form of a RangedSummarizedExperiment object and is
hosted on Bioconductor.</p>
<div class="section level3">
<h3 id="get-airway-data-from-package">Get <code>airway</code> data from package<a class="anchor" aria-label="anchor" href="#get-airway-data-from-package"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"airway"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"airway"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="get-counts-and-sample-annotation">Get counts and sample annotation<a class="anchor" aria-label="anchor" href="#get-counts-and-sample-annotation"></a>
</h3>
<p>The gene expression data is represented in counts. There are 4
samples treated with dexamethasone (<code>anno$dex == "trt"</code>) and
4 untreated samples (<code>anno$dex == "untrt"</code>).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Counts</span></span>
<span><span class="va">cts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">airway</span>, <span class="st">"counts"</span><span class="op">)</span></span>
<span><span class="co"># Annotation</span></span>
<span><span class="va">anno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">anno</span><span class="op">$</span><span class="va">dex</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/relevel.html" class="external-link">relevel</a></span><span class="op">(</span><span class="va">anno</span><span class="op">$</span><span class="va">dex</span>, <span class="st">"untrt"</span><span class="op">)</span> <span class="co"># Set untreated as first factor</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cts</span><span class="op">)</span></span>
<span><span class="co">#&gt;                 SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516</span></span>
<span><span class="co">#&gt; ENSG00000000003        679        448        873        408       1138</span></span>
<span><span class="co">#&gt; ENSG00000000005          0          0          0          0          0</span></span>
<span><span class="co">#&gt; ENSG00000000419        467        515        621        365        587</span></span>
<span><span class="co">#&gt; ENSG00000000457        260        211        263        164        245</span></span>
<span><span class="co">#&gt; ENSG00000000460         60         55         40         35         78</span></span>
<span><span class="co">#&gt; ENSG00000000938          0          0          2          0          1</span></span>
<span><span class="co">#&gt;                 SRR1039517 SRR1039520 SRR1039521</span></span>
<span><span class="co">#&gt; ENSG00000000003       1047        770        572</span></span>
<span><span class="co">#&gt; ENSG00000000005          0          0          0</span></span>
<span><span class="co">#&gt; ENSG00000000419        799        417        508</span></span>
<span><span class="co">#&gt; ENSG00000000457        331        233        229</span></span>
<span><span class="co">#&gt; ENSG00000000460         63         76         60</span></span>
<span><span class="co">#&gt; ENSG00000000938          0          0          0</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">anno</span><span class="op">)</span></span>
<span><span class="co">#&gt;            SampleName    cell   dex albut        Run avgLength Experiment</span></span>
<span><span class="co">#&gt; SRR1039508 GSM1275862  N61311 untrt untrt SRR1039508       126  SRX384345</span></span>
<span><span class="co">#&gt; SRR1039509 GSM1275863  N61311   trt untrt SRR1039509       126  SRX384346</span></span>
<span><span class="co">#&gt; SRR1039512 GSM1275866 N052611 untrt untrt SRR1039512       126  SRX384349</span></span>
<span><span class="co">#&gt; SRR1039513 GSM1275867 N052611   trt untrt SRR1039513        87  SRX384350</span></span>
<span><span class="co">#&gt; SRR1039516 GSM1275870 N080611 untrt untrt SRR1039516       120  SRX384353</span></span>
<span><span class="co">#&gt; SRR1039517 GSM1275871 N080611   trt untrt SRR1039517       126  SRX384354</span></span>
<span><span class="co">#&gt;               Sample    BioSample</span></span>
<span><span class="co">#&gt; SRR1039508 SRS508568 SAMN02422669</span></span>
<span><span class="co">#&gt; SRR1039509 SRS508567 SAMN02422675</span></span>
<span><span class="co">#&gt; SRR1039512 SRS508571 SAMN02422678</span></span>
<span><span class="co">#&gt; SRR1039513 SRS508572 SAMN02422670</span></span>
<span><span class="co">#&gt; SRR1039516 SRS508575 SAMN02422682</span></span>
<span><span class="co">#&gt; SRR1039517 SRS508576 SAMN02422673</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="visualize-samples-w-pca">Visualize samples w/ PCA<a class="anchor" aria-label="anchor" href="#visualize-samples-w-pca"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># log2UQ</span></span>
<span><span class="co"># Rubrary::run_PCA</span></span>
<span><span class="co"># Rubrary::plot_PCA</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="deseq2-differentially-expressed-genes">
<code>DESeq2</code> differentially expressed genes<a class="anchor" aria-label="anchor" href="#deseq2-differentially-expressed-genes"></a>
</h2>
<p><code>DESeq2</code> <span class="citation">[2]</span> performs
differential gene expression analysis my modeling counts as a negative
binomial distribution.</p>
<div class="section level3">
<h3 id="create-deseqdataset-object-and-run-deseq">Create DESeqDataSet object and run DESeq<a class="anchor" aria-label="anchor" href="#create-deseqdataset-object-and-run-deseq"></a>
</h3>
<p><code><a href="../reference/output_DESeq.html">Rubrary::output_DESeq</a></code> takes the DESeqResults object,
transforms it to a dataframe, and adds a sign log2 p-value column.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mikelove/DESeq2" class="external-link">DESeq2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSetFromMatrix</a></span><span class="op">(</span></span>
<span>  countData <span class="op">=</span> <span class="va">cts</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">anno</span><span class="op">)</span><span class="op">]</span>,</span>
<span>  colData <span class="op">=</span> <span class="va">anno</span>,</span>
<span>  design <span class="op">=</span> <span class="op">~</span> <span class="va">dex</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span>
<span><span class="co">#&gt; estimating size factors</span></span>
<span><span class="co">#&gt; estimating dispersions</span></span>
<span><span class="co">#&gt; gene-wise dispersion estimates</span></span>
<span><span class="co">#&gt; mean-dispersion relationship</span></span>
<span><span class="co">#&gt; final dispersion estimates</span></span>
<span><span class="co">#&gt; fitting model and testing</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">results</a></span><span class="op">(</span><span class="va">dds</span>, contrast <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"dex"</span>, <span class="st">"trt"</span>, <span class="st">"untrt"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">deseq_res_df</span> <span class="op">&lt;-</span> <span class="fu">Rubrary</span><span class="fu">::</span><span class="fu"><a href="../reference/output_DESeq.html">output_DESeq</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="replace-ensembl-id-w-hgnc-symbol">Replace Ensembl ID w/ HGNC symbol<a class="anchor" aria-label="anchor" href="#replace-ensembl-id-w-hgnc-symbol"></a>
</h4>
<p><code>airway</code> gene names are given in <a href="http://www.ensembl.org/info/genome/stable_ids/index.html" class="external-link">Ensembl
ID</a>, but <a href="https://www.genenames.org/about/guidelines/" class="external-link">HGNC
gene symbols</a> can be easier to interpret in downstream analysis and
more compatible with gene set enrichment analysis (GSEA) tools.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Gene format is currently in Ensembl ID format</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">deseq_res_df</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "ENSG00000152583" "ENSG00000148175" "ENSG00000179094" "ENSG00000134686"</span></span>
<span><span class="co">#&gt; [5] "ENSG00000125148" "ENSG00000120129"</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/grimbough/biomaRt" class="external-link">biomaRt</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'utils::findMatches' by</span></span>
<span><span class="co">#&gt; 'S4Vectors::findMatches' when loading 'AnnotationDbi'</span></span>
<span><span class="co"># Load Mart object - database of Homo sapiens genes from Ensembl</span></span>
<span><span class="va">mart</span> <span class="op">&lt;-</span> <span class="fu">biomaRt</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useDataset.html" class="external-link">useDataset</a></span><span class="op">(</span></span>
<span>  dataset <span class="op">=</span> <span class="st">"hsapiens_gene_ensembl"</span>,</span>
<span>  mart <span class="op">=</span> <span class="fu">biomaRt</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useMart.html" class="external-link">useMart</a></span><span class="op">(</span><span class="st">"ENSEMBL_MART_ENSEMBL"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">conv</span> <span class="op">&lt;-</span> <span class="fu">Rubrary</span><span class="fu">::</span><span class="fu"><a href="../reference/convert_genes.html">convert_genes</a></span><span class="op">(</span></span>
<span>    genes <span class="op">=</span> <span class="va">deseq_res_df</span><span class="op">$</span><span class="va">gene</span>,</span>
<span>    from_to <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ensembl_gene_id"</span>, <span class="st">"hgnc_symbol"</span><span class="op">)</span>,</span>
<span>    mart <span class="op">=</span> <span class="va">mart</span>, table <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">hgnc_symbol</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="co"># Exclude unmatched HGNC symbols</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="va">hgnc_symbol</span>, .keep_all <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="va">deseq_res_hgnc</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span> <span class="co"># Join stats with HGNC symbol table</span></span>
<span>  <span class="va">conv</span>, <span class="va">deseq_res_df</span>, by <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/join_by.html" class="external-link">join_by</a></span><span class="op">(</span><span class="va">ensembl_gene_id</span> <span class="op">==</span> <span class="va">gene</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/MaskCollection-class.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">sign_log_p</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Genes now have associated HGNC symbols</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">deseq_res_hgnc</span><span class="op">)</span></span>
<span><span class="co">#&gt;   ensembl_gene_id hgnc_symbol   baseMean log2FoldChange      lfcSE     stat</span></span>
<span><span class="co">#&gt; 1 ENSG00000152583     SPARCL1   997.4398       4.602526 0.21177076 21.73353</span></span>
<span><span class="co">#&gt; 2 ENSG00000148175        STOM 11193.7188       1.451466 0.08482489 17.11132</span></span>
<span><span class="co">#&gt; 3 ENSG00000179094        PER1   776.5967       3.183858 0.20151544 15.79957</span></span>
<span><span class="co">#&gt; 4 ENSG00000134686        PHC2  2737.9820       1.387141 0.09158425 15.14607</span></span>
<span><span class="co">#&gt; 5 ENSG00000125148        MT2A  3656.2528       2.203439 0.14740866 14.94782</span></span>
<span><span class="co">#&gt; 6 ENSG00000120129       DUSP1  3409.0294       2.948984 0.20161362 14.62691</span></span>
<span><span class="co">#&gt;          pvalue          padj sign_log_p</span></span>
<span><span class="co">#&gt; 1 9.889991e-105 1.838055e-100   345.4965</span></span>
<span><span class="co">#&gt; 2  1.221959e-65  1.135505e-61   215.6361</span></span>
<span><span class="co">#&gt; 3  3.132381e-56  1.940510e-52   184.3807</span></span>
<span><span class="co">#&gt; 4  8.043913e-52  3.737403e-48   169.7324</span></span>
<span><span class="co">#&gt; 5  1.609198e-50  5.981389e-47   165.4101</span></span>
<span><span class="co">#&gt; 6  1.891923e-48  5.860231e-45   158.5327</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="filter-to-protein-coding-genes">Filter to protein coding genes<a class="anchor" aria-label="anchor" href="#filter-to-protein-coding-genes"></a>
</h3>
<p>For ease of interpretation, the DE genes results can also be filtered
to protein coding genes only. <code><a href="../reference/filter_genes.html">Rubrary::filter_genes()</a></code> can
filter with the list of protein coding genes retrieved with
<code><a href="../reference/get_PC_genes.html">Rubrary::get_PC_genes()</a></code>. Since we already created the
<code>Mart</code> object when converting gene symbols, we can reuse it
to avoid querying the BioMart database again and load protein coding
genes efficiently.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">deseq_res_hgnc_PC</span> <span class="op">&lt;-</span> <span class="fu">Rubrary</span><span class="fu">::</span><span class="fu"><a href="../reference/filter_genes.html">filter_genes</a></span><span class="op">(</span></span>
<span>  df <span class="op">=</span> <span class="va">deseq_res_hgnc</span>,</span>
<span>  gene_col <span class="op">=</span> <span class="st">"hgnc_symbol"</span>,</span>
<span>  genes_filt <span class="op">=</span> <span class="fu">Rubrary</span><span class="fu">::</span><span class="fu"><a href="../reference/get_PC_genes.html">get_PC_genes</a></span><span class="op">(</span><span class="va">mart</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="volcano-plot">Volcano plot<a class="anchor" aria-label="anchor" href="#volcano-plot"></a>
</h3>
<p>DESeq results can be visualized through volcano plots. Volcano plots
show that top and bottom differentially expressed genes when sorted by
decreasing signed log p value present as outliers on the top right and
top left.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># All DE genes</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">deseq_res_hgnc</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">deseq_res_hgnc</span><span class="op">$</span><span class="va">sign_log_p</span>, decreasing <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>,<span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;   ensembl_gene_id hgnc_symbol   baseMean log2FoldChange      lfcSE     stat</span></span>
<span><span class="co">#&gt; 1 ENSG00000152583     SPARCL1   997.4398       4.602526 0.21177076 21.73353</span></span>
<span><span class="co">#&gt; 2 ENSG00000148175        STOM 11193.7188       1.451466 0.08482489 17.11132</span></span>
<span><span class="co">#&gt; 3 ENSG00000179094        PER1   776.5967       3.183858 0.20151544 15.79957</span></span>
<span><span class="co">#&gt; 4 ENSG00000134686        PHC2  2737.9820       1.387141 0.09158425 15.14607</span></span>
<span><span class="co">#&gt; 5 ENSG00000125148        MT2A  3656.2528       2.203439 0.14740866 14.94782</span></span>
<span><span class="co">#&gt; 6 ENSG00000120129       DUSP1  3409.0294       2.948984 0.20161362 14.62691</span></span>
<span><span class="co">#&gt;          pvalue          padj sign_log_p</span></span>
<span><span class="co">#&gt; 1 9.889991e-105 1.838055e-100   345.4965</span></span>
<span><span class="co">#&gt; 2  1.221959e-65  1.135505e-61   215.6361</span></span>
<span><span class="co">#&gt; 3  3.132381e-56  1.940510e-52   184.3807</span></span>
<span><span class="co">#&gt; 4  8.043913e-52  3.737403e-48   169.7324</span></span>
<span><span class="co">#&gt; 5  1.609198e-50  5.981389e-47   165.4101</span></span>
<span><span class="co">#&gt; 6  1.891923e-48  5.860231e-45   158.5327</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="va">deseq_res_hgnc</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">deseq_res_hgnc</span><span class="op">$</span><span class="va">sign_log_p</span>, decreasing <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>,<span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;       ensembl_gene_id hgnc_symbol    baseMean log2FoldChange      lfcSE</span></span>
<span><span class="co">#&gt; 25081 ENSG00000162692       VCAM1    508.1702      -3.609367 0.32654691</span></span>
<span><span class="co">#&gt; 25082 ENSG00000108821      COL1A1 100990.6944      -1.299499 0.11528151</span></span>
<span><span class="co">#&gt; 25083 ENSG00000144369     FAM171B   1303.2727      -1.350324 0.11607419</span></span>
<span><span class="co">#&gt; 25084 ENSG00000116584     ARHGEF2   2250.5969      -1.014517 0.08232161</span></span>
<span><span class="co">#&gt; 25085 ENSG00000196517      SLC6A9    229.6407      -2.216934 0.16666134</span></span>
<span><span class="co">#&gt; 25086 ENSG00000178695      KCTD12   2649.8501      -2.499912 0.17893290</span></span>
<span><span class="co">#&gt;            stat       pvalue         padj sign_log_p</span></span>
<span><span class="co">#&gt; 25081 -11.05314 2.116887e-28 1.639264e-25  -91.93204</span></span>
<span><span class="co">#&gt; 25082 -11.27239 1.796180e-29 1.589619e-26  -95.49098</span></span>
<span><span class="co">#&gt; 25083 -11.63328 2.791614e-31 2.882341e-28 -101.49867</span></span>
<span><span class="co">#&gt; 25084 -12.32382 6.742097e-35 8.353458e-32 -113.51429</span></span>
<span><span class="co">#&gt; 25085 -13.30203 2.252620e-40 3.488745e-37 -131.70552</span></span>
<span><span class="co">#&gt; 25086 -13.97123 2.335645e-44 4.823106e-41 -144.94102</span></span>
<span></span>
<span><span class="fu">Rubrary</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_volcano.html">plot_volcano</a></span><span class="op">(</span></span>
<span>  df_deg <span class="op">=</span> <span class="va">deseq_res_hgnc</span>,</span>
<span>  names <span class="op">=</span> <span class="st">"hgnc_symbol"</span>,</span>
<span>  x <span class="op">=</span> <span class="st">"log2FoldChange"</span>, y <span class="op">=</span> <span class="st">"pvalue"</span>,</span>
<span>  FCcutoff <span class="op">=</span> <span class="fl">1</span>, pCutoff <span class="op">=</span> <span class="fl">5e-04</span>,</span>
<span>  title <span class="op">=</span> <span class="st">"Airway Treated vs. Untreated DE Genes"</span>,</span>
<span>  xlab_high <span class="op">=</span> <span class="st">"\U2191 in treated,\n\U2193 in untreated"</span>,</span>
<span>  xlab_low <span class="op">=</span> <span class="st">"\U2191 in untreated,\n\U2193 in treated"</span>,</span>
<span>  subtitle <span class="op">=</span> <span class="st">"All genes"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="DESeq_GSEA_files/figure-html/Volcano_plot-1.png" width="1536"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Protein coding DE genes</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">deseq_res_hgnc_PC</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">deseq_res_hgnc</span><span class="op">$</span><span class="va">sign_log_p</span>, decreasing <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>,<span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;   ensembl_gene_id hgnc_symbol   baseMean log2FoldChange      lfcSE     stat</span></span>
<span><span class="co">#&gt; 1 ENSG00000152583     SPARCL1   997.4398       4.602526 0.21177076 21.73353</span></span>
<span><span class="co">#&gt; 2 ENSG00000148175        STOM 11193.7188       1.451466 0.08482489 17.11132</span></span>
<span><span class="co">#&gt; 3 ENSG00000179094        PER1   776.5967       3.183858 0.20151544 15.79957</span></span>
<span><span class="co">#&gt; 4 ENSG00000134686        PHC2  2737.9820       1.387141 0.09158425 15.14607</span></span>
<span><span class="co">#&gt; 5 ENSG00000125148        MT2A  3656.2528       2.203439 0.14740866 14.94782</span></span>
<span><span class="co">#&gt; 6 ENSG00000120129       DUSP1  3409.0294       2.948984 0.20161362 14.62691</span></span>
<span><span class="co">#&gt;          pvalue          padj sign_log_p</span></span>
<span><span class="co">#&gt; 1 9.889991e-105 1.838055e-100   345.4965</span></span>
<span><span class="co">#&gt; 2  1.221959e-65  1.135505e-61   215.6361</span></span>
<span><span class="co">#&gt; 3  3.132381e-56  1.940510e-52   184.3807</span></span>
<span><span class="co">#&gt; 4  8.043913e-52  3.737403e-48   169.7324</span></span>
<span><span class="co">#&gt; 5  1.609198e-50  5.981389e-47   165.4101</span></span>
<span><span class="co">#&gt; 6  1.891923e-48  5.860231e-45   158.5327</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="va">deseq_res_hgnc_PC</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">deseq_res_hgnc</span><span class="op">$</span><span class="va">sign_log_p</span>, decreasing <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>,<span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;         ensembl_gene_id hgnc_symbol baseMean log2FoldChange lfcSE stat pvalue</span></span>
<span><span class="co">#&gt; NA.8620            &lt;NA&gt;        &lt;NA&gt;       NA             NA    NA   NA     NA</span></span>
<span><span class="co">#&gt; NA.8621            &lt;NA&gt;        &lt;NA&gt;       NA             NA    NA   NA     NA</span></span>
<span><span class="co">#&gt; NA.8622            &lt;NA&gt;        &lt;NA&gt;       NA             NA    NA   NA     NA</span></span>
<span><span class="co">#&gt; NA.8623            &lt;NA&gt;        &lt;NA&gt;       NA             NA    NA   NA     NA</span></span>
<span><span class="co">#&gt; NA.8624            &lt;NA&gt;        &lt;NA&gt;       NA             NA    NA   NA     NA</span></span>
<span><span class="co">#&gt; NA.8625            &lt;NA&gt;        &lt;NA&gt;       NA             NA    NA   NA     NA</span></span>
<span><span class="co">#&gt;         padj sign_log_p</span></span>
<span><span class="co">#&gt; NA.8620   NA         NA</span></span>
<span><span class="co">#&gt; NA.8621   NA         NA</span></span>
<span><span class="co">#&gt; NA.8622   NA         NA</span></span>
<span><span class="co">#&gt; NA.8623   NA         NA</span></span>
<span><span class="co">#&gt; NA.8624   NA         NA</span></span>
<span><span class="co">#&gt; NA.8625   NA         NA</span></span>
<span></span>
<span><span class="fu">Rubrary</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_volcano.html">plot_volcano</a></span><span class="op">(</span></span>
<span>  df_deg <span class="op">=</span> <span class="va">deseq_res_hgnc_PC</span>,</span>
<span>  names <span class="op">=</span> <span class="st">"hgnc_symbol"</span>,</span>
<span>  x <span class="op">=</span> <span class="st">"log2FoldChange"</span>, y <span class="op">=</span> <span class="st">"pvalue"</span>,</span>
<span>  FCcutoff <span class="op">=</span> <span class="fl">1</span>, pCutoff <span class="op">=</span> <span class="fl">5e-04</span>,</span>
<span>  title <span class="op">=</span> <span class="st">"Airway Treated vs. Untreated DE Genes"</span>,</span>
<span>  xlab_high <span class="op">=</span> <span class="st">"\U2191 in treated\n\U2193 in untreated"</span>,</span>
<span>  xlab_low <span class="op">=</span> <span class="st">"\U2191 in untreated\n\U2193 in treated"</span>,</span>
<span>  subtitle <span class="op">=</span> <span class="st">"Protein coding genes"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="DESeq_GSEA_files/figure-html/Volcano_plot-2.png" width="1536"></p>
</div>
</div>
<div class="section level2">
<h2 id="fgsea-gene-set-enrichment-analysis">
<code>fgsea</code> gene set enrichment analysis<a class="anchor" aria-label="anchor" href="#fgsea-gene-set-enrichment-analysis"></a>
</h2>
<p><code>fgsea</code> <span class="citation">[3]</span> is an
implementation of an algorithm for fast gene set enrichment analysis
(GSEA), based on the original GSEA algorithm <span class="citation">[4]</span>.</p>
<div class="section level3">
<h3 id="get-pathways-from-msigdbr">Get pathways from <code>msigdbr</code><a class="anchor" aria-label="anchor" href="#get-pathways-from-msigdbr"></a>
</h3>
<p>Gene set enrichment analysis relies on inputting gene sets / pathways
related to biological function. The Molecular Signatures Database
(MSigDB) <span class="citation">[5]</span> was published as a database
of annotated pathways to facilitate GSEA and is accessible through the R
package <code>msigdbr</code> <span class="citation">[6]</span>.</p>
<p><a href="https://www.gsea-msigdb.org/gsea/msigdb/human/collection_details.jsp" class="external-link">Human
MSigDB Collections</a></p>
<ul>
<li><p>C2CP: curated gene sets, canonical pathways</p></li>
<li>
<p>C5GO: ontology gene sets, Gene Ontology gene sets</p>
<ul>
<li><p>BP: GO Biological Process ontology</p></li>
<li><p>CC: GO Cellular Component ontology</p></li>
<li><p>MF: GO Molecular Function ontology</p></li>
</ul>
</li>
<li><p>H: hallmark gene sets <span class="citation">[7]</span></p></li>
</ul>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"msigdbr"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"msigdbr"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">msigdbr</span><span class="fu">::</span><span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr_collections.html" class="external-link">msigdbr_collections</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 23 × 3</span></span></span>
<span><span class="co">#&gt;    gs_cat gs_subcat         num_genesets</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> C1     <span style="color: #949494;">""</span>                         299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> C2     <span style="color: #949494;">"</span>CGP<span style="color: #949494;">"</span>                     <span style="text-decoration: underline;">3</span>384</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> C2     <span style="color: #949494;">"</span>CP<span style="color: #949494;">"</span>                        29</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> C2     <span style="color: #949494;">"</span>CP:BIOCARTA<span style="color: #949494;">"</span>              292</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> C2     <span style="color: #949494;">"</span>CP:KEGG<span style="color: #949494;">"</span>                  186</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> C2     <span style="color: #949494;">"</span>CP:PID<span style="color: #949494;">"</span>                   196</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> C2     <span style="color: #949494;">"</span>CP:REACTOME<span style="color: #949494;">"</span>             <span style="text-decoration: underline;">1</span>615</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> C2     <span style="color: #949494;">"</span>CP:WIKIPATHWAYS<span style="color: #949494;">"</span>          664</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> C3     <span style="color: #949494;">"</span>MIR:MIRDB<span style="color: #949494;">"</span>               <span style="text-decoration: underline;">2</span>377</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> C3     <span style="color: #949494;">"</span>MIR:MIR_Legacy<span style="color: #949494;">"</span>           221</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 13 more rows</span></span></span>
<span></span>
<span><span class="va">pthwys</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span></span>
<span>  <span class="fu">msigdbr</span><span class="fu">::</span><span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr.html" class="external-link">msigdbr</a></span><span class="op">(</span>category <span class="op">=</span> <span class="st">"C2"</span>, subcategory <span class="op">=</span> <span class="st">"CP"</span><span class="op">)</span>,</span>
<span>  <span class="fu">msigdbr</span><span class="fu">::</span><span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr.html" class="external-link">msigdbr</a></span><span class="op">(</span>category <span class="op">=</span> <span class="st">"C5"</span>, subcategory <span class="op">=</span> <span class="st">"GO:BP"</span><span class="op">)</span>,</span>
<span>  <span class="fu">msigdbr</span><span class="fu">::</span><span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr.html" class="external-link">msigdbr</a></span><span class="op">(</span>category <span class="op">=</span> <span class="st">"C5"</span>, subcategory <span class="op">=</span> <span class="st">"GO:CC"</span><span class="op">)</span>,</span>
<span>  <span class="fu">msigdbr</span><span class="fu">::</span><span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr.html" class="external-link">msigdbr</a></span><span class="op">(</span>category <span class="op">=</span> <span class="st">"C5"</span>, subcategory <span class="op">=</span> <span class="st">"GO:MF"</span><span class="op">)</span>,</span>
<span>  <span class="fu">msigdbr</span><span class="fu">::</span><span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr.html" class="external-link">msigdbr</a></span><span class="op">(</span>category <span class="op">=</span> <span class="st">"H"</span><span class="op">)</span></span>
<span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">split</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">.</span><span class="op">$</span><span class="va">gene_symbol</span>, f <span class="op">=</span> <span class="va">.</span><span class="op">$</span><span class="va">gs_name</span><span class="op">)</span> <span class="co"># Named list of pathways</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="run-fgsea">Run <code>fgsea</code><a class="anchor" aria-label="anchor" href="#run-fgsea"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># `fgsea` requires a named numeric vector as input to the `stats` argument</span></span>
<span><span class="va">deseq_stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span></span>
<span>    <span class="va">deseq_res_hgnc_PC</span><span class="op">[</span>,<span class="st">"sign_log_p"</span><span class="op">]</span>, </span>
<span>    <span class="va">deseq_res_hgnc_PC</span><span class="op">[</span>,<span class="st">"hgnc_symbol"</span><span class="op">]</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">gsea_results</span> <span class="op">&lt;-</span> <span class="fu">fgsea</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/fgsea/man/fgsea.html" class="external-link">fgsea</a></span><span class="op">(</span></span>
<span>  pathways <span class="op">=</span> <span class="va">pthwys</span>,</span>
<span>  stats <span class="op">=</span> <span class="va">deseq_stats</span>,</span>
<span>  eps <span class="op">=</span> <span class="fl">0.0</span>,</span>
<span>  minSize <span class="op">=</span> <span class="fl">15</span>,</span>
<span>  maxSize  <span class="op">=</span> <span class="fl">500</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/MaskCollection-class.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">NES</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>sign_log10_p <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sign.html" class="external-link">sign</a></span><span class="op">(</span><span class="va">NES</span><span class="op">)</span> <span class="op">*</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">pval</span><span class="op">)</span><span class="op">)</span> <span class="co"># Create sign_log10_p column</span></span>
<span><span class="co">#&gt; Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (6.92% of the list).</span></span>
<span><span class="co">#&gt; The order of those tied genes will be arbitrary, which may produce unexpected results.</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="visualize-gsea-results-as-a-table">Visualize GSEA results as a table<a class="anchor" aria-label="anchor" href="#visualize-gsea-results-as-a-table"></a>
</h3>
<p>See the <a href="https://www.gsea-msigdb.org/gsea/doc/GSEAUserGuideTEXT.htm#_Interpreting_GSEA_Results" class="external-link">GSEA
User Guide: Interpreting GSEA Results</a> for detailed explanation in
interpreting results.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get top significant pathways by positive/negative enrichment score (ES)</span></span>
<span><span class="va">pos_pws</span> <span class="op">&lt;-</span> <span class="va">gsea_results</span><span class="op">[</span><span class="va">ES</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">]</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">pval</span><span class="op">)</span>, n <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>, <span class="va">pathway</span><span class="op">]</span></span>
<span><span class="va">neg_pws</span> <span class="op">&lt;-</span> <span class="va">gsea_results</span><span class="op">[</span><span class="va">ES</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">]</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">pval</span><span class="op">)</span>, n <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>, <span class="va">pathway</span><span class="op">]</span></span>
<span><span class="va">top_pws</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">pos_pws</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">rev</a></span><span class="op">(</span><span class="va">neg_pws</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">fgsea</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/fgsea/man/plotGseaTable.html" class="external-link">plotGseaTable</a></span><span class="op">(</span></span>
<span>  pathways <span class="op">=</span> <span class="va">pthwys</span><span class="op">[</span><span class="va">top_pws</span><span class="op">]</span>,</span>
<span>  stats <span class="op">=</span> <span class="va">deseq_stats</span>,</span>
<span>  fgseaRes <span class="op">=</span> <span class="va">gsea_results</span>,</span>
<span>  gseaParam <span class="op">=</span> <span class="fl">0.5</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="DESeq_GSEA_files/figure-html/GSEA_table-1.png" width="1152"></p>
</div>
<div class="section level3">
<h3 id="pathway-enrichment-plot">Pathway enrichment plot<a class="anchor" aria-label="anchor" href="#pathway-enrichment-plot"></a>
</h3>
<p>Specific pathways of interest can be plotted in GSEA Java app-like
enrichment plots that combine a waterfall and mountain plot.</p>
<p>The top enriched pathway from our GSEA results is “<a href="https://www.gsea-msigdb.org/gsea/msigdb/human/geneset/HALLMARK_ADIPOGENESIS.html" class="external-link">HALLMARK_ADIPOGENESIS</a>”,
a collection of 196 genes up-regulated during adipocyte (fat cell)
differentiation. This is in line with studies reporting the complex
effects of glucocorticoids on adipose tissue biology, including
differentiation of adipocyte precursors and adipogenesis <span class="citation">[8,9]</span>.</p>
<p>The maximum absolute value in the Enrichment Score mountain plot
corresponds to the <code>ES</code> value reported in the GSEA
results.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gsea_results</span><span class="op">[</span><span class="va">gsea_results</span><span class="op">$</span><span class="va">pathway</span> <span class="op">==</span> <span class="st">"HALLMARK_ADIPOGENESIS"</span>,<span class="op">]</span></span>
<span><span class="co">#&gt;                  pathway         pval         padj   log2err        ES      NES</span></span>
<span><span class="co">#&gt; 1: HALLMARK_ADIPOGENESIS 4.505836e-07 0.0007826638 0.6749629 0.6955378 1.919844</span></span>
<span><span class="co">#&gt;    size                           leadingEdge sign_log10_p</span></span>
<span><span class="co">#&gt; 1:  196 SPARCL1,STOM,GPX3,TST,SLC5A6,SQOR,...     6.346225</span></span>
<span></span>
<span><span class="fu">Rubrary</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_GSEA_pathway.html">plot_GSEA_pathway</a></span><span class="op">(</span></span>
<span>  sig <span class="op">=</span> <span class="va">deseq_res_hgnc_PC</span>,</span>
<span>  geneset <span class="op">=</span> <span class="va">pthwys</span><span class="op">[[</span><span class="st">"HALLMARK_ADIPOGENESIS"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  genecol <span class="op">=</span> <span class="st">"hgnc_symbol"</span>,</span>
<span>  rankcol <span class="op">=</span> <span class="st">"sign_log_p"</span>,</span>
<span>  rankcol_name <span class="op">=</span> <span class="st">"DESeq sign log2 p-value"</span>,</span>
<span>  label <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  title <span class="op">=</span> <span class="st">"Airway DE Genes: HALLMARK_ADIPOGENESIS"</span>,</span>
<span>  lab_high <span class="op">=</span> <span class="st">"\U2191 in treated\n\U2193 in untreated"</span>,</span>
<span>  lab_low <span class="op">=</span> <span class="st">"\U2191 in untreated\n\U2193 in treated"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="DESeq_GSEA_files/figure-html/GSEA_pathway-1.png" width="768"></p>
<p>For gene sets of smaller sizes, like 20 gene pathway “<a href="https://www.gsea-msigdb.org/gsea/msigdb/human/geneset/GOBP_NEURON_FATE_SPECIFICATION.html" class="external-link">GOBP_NEURON_FATE_SPECIFICATION</a>”,
labeling each gene may be appropriate.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gsea_results</span><span class="op">[</span><span class="va">gsea_results</span><span class="op">$</span><span class="va">pathway</span> <span class="op">==</span> <span class="st">"GOBP_NEURON_FATE_SPECIFICATION"</span>,<span class="op">]</span></span>
<span><span class="co">#&gt;                           pathway         pval       padj   log2err         ES</span></span>
<span><span class="co">#&gt; 1: GOBP_NEURON_FATE_SPECIFICATION 3.084947e-05 0.01786184 0.5573322 -0.8621391</span></span>
<span><span class="co">#&gt;          NES size               leadingEdge sign_log10_p</span></span>
<span><span class="co">#&gt; 1: -1.888097   20 GLI2,SUFU,GLI3,EYA1,EHMT2    -4.510752</span></span>
<span></span>
<span><span class="fu">Rubrary</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_GSEA_pathway.html">plot_GSEA_pathway</a></span><span class="op">(</span></span>
<span>  sig <span class="op">=</span> <span class="va">deseq_res_hgnc_PC</span>,</span>
<span>  geneset <span class="op">=</span> <span class="va">pthwys</span><span class="op">[[</span><span class="st">"GOBP_NEURON_FATE_SPECIFICATION"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  genecol <span class="op">=</span> <span class="st">"hgnc_symbol"</span>,</span>
<span>  rankcol <span class="op">=</span> <span class="st">"sign_log_p"</span>,</span>
<span>  rankcol_name <span class="op">=</span> <span class="st">"DESeq sign log2 p-value"</span>,</span>
<span>  label <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  title <span class="op">=</span> <span class="st">"Airway DE Genes: GOBP_NEURON_FATE_SPECIFICATION"</span>,</span>
<span>  lab_high <span class="op">=</span> <span class="st">"\U2191 in treated\n\U2193 in untreated"</span>,</span>
<span>  lab_low <span class="op">=</span> <span class="st">"\U2191 in untreated\n\U2193 in treated"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 16440 rows containing missing values</span></span>
<span><span class="co">#&gt; (`geom_text_repel()`).</span></span></code></pre></div>
<p><img src="DESeq_GSEA_files/figure-html/GSEA_pathway_labeled-1.png" width="768"></p>
<p>Multiple gene set enrichment plots can be created in a batch with
<code><a href="../reference/plot_GSEA_batch.html">Rubrary::plot_GSEA_batch</a></code>, a wrapper for
<code><a href="../reference/plot_GSEA_pathway.html">Rubrary::plot_GSEA_pathway</a></code> that works well with
<code>lapply</code>. We can use the <code>neg_pws</code> list of pathway
names to plot gene enrichment plots for the most significant negatively
enriched pathways.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gsea_results</span><span class="op">[</span><span class="va">gsea_results</span><span class="op">$</span><span class="va">pathway</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">neg_pws</span>,<span class="op">]</span></span>
<span><span class="co">#&gt;                                              pathway         pval         padj</span></span>
<span><span class="co">#&gt; 1:        GOMF_SIGNALING_RECEPTOR_REGULATOR_ACTIVITY 7.663365e-06 0.0099834492</span></span>
<span><span class="co">#&gt; 2:                            GOCC_SYNAPTIC_MEMBRANE 2.676290e-07 0.0006973074</span></span>
<span><span class="co">#&gt; 3:                        GOCC_POSTSYNAPTIC_MEMBRANE 2.336522e-07 0.0006973074</span></span>
<span><span class="co">#&gt; 4:     GOCC_INTRINSIC_COMPONENT_OF_SYNAPTIC_MEMBRANE 1.445414e-05 0.0125534240</span></span>
<span><span class="co">#&gt; 5: GOCC_INTRINSIC_COMPONENT_OF_POSTSYNAPTIC_MEMBRANE 1.331605e-05 0.0125534240</span></span>
<span><span class="co">#&gt;      log2err         ES       NES size</span></span>
<span><span class="co">#&gt; 1: 0.5933255 -0.4721901 -1.602508  387</span></span>
<span><span class="co">#&gt; 2: 0.6749629 -0.5283197 -1.765797  319</span></span>
<span><span class="co">#&gt; 3: 0.6901325 -0.5817831 -1.879653  226</span></span>
<span><span class="co">#&gt; 4: 0.5933255 -0.6182693 -1.889896  128</span></span>
<span><span class="co">#&gt; 5: 0.5933255 -0.6684340 -1.932328   98</span></span>
<span><span class="co">#&gt;                                    leadingEdge sign_log10_p</span></span>
<span><span class="co">#&gt; 1:        TSLP,CCN3,LIF,CXCL12,FLRT3,GDF15,...    -5.115580</span></span>
<span><span class="co">#&gt; 2:  KCTD12,SLC6A9,SLC6A6,LRRTM2,DNM1,FLRT3,...    -6.572467</span></span>
<span><span class="co">#&gt; 3: KCTD12,SLC6A9,SLC6A6,LRRTM2,FLRT3,CNIH3,...    -6.631430</span></span>
<span><span class="co">#&gt; 4:  SLC6A9,SLC6A6,LRRTM2,FLRT3,ADCY8,EPHB2,...    -4.840008</span></span>
<span><span class="co">#&gt; 5: SLC6A9,SLC6A6,LRRTM2,FLRT3,EPHB2,ADRA2A,...    -4.875625</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span></span>
<span>  <span class="va">neg_pws</span>,</span>
<span>  FUN <span class="op">=</span> <span class="fu">Rubrary</span><span class="fu">::</span><span class="va"><a href="../reference/plot_GSEA_batch.html">plot_GSEA_batch</a></span>,</span>
<span>  genecol <span class="op">=</span> <span class="st">"hgnc_symbol"</span>,</span>
<span>  pthwys <span class="op">=</span> <span class="va">pthwys</span>,</span>
<span>  sig <span class="op">=</span> <span class="va">deseq_res_hgnc_PC</span>,</span>
<span>  rankcol <span class="op">=</span> <span class="st">"sign_log_p"</span>,</span>
<span>  rankcol_name <span class="op">=</span> <span class="st">"DESeq sign log2 p-value"</span>,</span>
<span>  hllab <span class="op">=</span> <span class="st">"Pathway genes"</span>,</span>
<span>  lab_high <span class="op">=</span> <span class="st">"\U2191 in treated\n\U2193 in untreated"</span>,</span>
<span>  lab_low <span class="op">=</span> <span class="st">"\U2191 in untreated\n\U2193 in treated"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; [[1]]</span></span></code></pre></div>
<p><img src="DESeq_GSEA_files/figure-html/GSEA_pathway_batch-1.png" width="768"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[2]]</span></span></code></pre>
<p><img src="DESeq_GSEA_files/figure-html/GSEA_pathway_batch-2.png" width="768"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[3]]</span></span></code></pre>
<p><img src="DESeq_GSEA_files/figure-html/GSEA_pathway_batch-3.png" width="768"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[4]]</span></span></code></pre>
<p><img src="DESeq_GSEA_files/figure-html/GSEA_pathway_batch-4.png" width="768"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[5]]</span></span></code></pre>
<p><img src="DESeq_GSEA_files/figure-html/GSEA_pathway_batch-5.png" width="768"></p>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="citations">Citations<a class="anchor" aria-label="anchor" href="#citations"></a>
</h2>
<div id="refs" class="references csl-bib-body">
<div id="ref-himesRNASeqTranscriptomeProfiling2014" class="csl-entry">
<div class="csl-left-margin">[1] </div>
<div class="csl-right-inline">Himes BE, Jiang X, Wagner P, Hu R, Wang Q,
Klanderman B, et al. <span>RNA-Seq Transcriptome Profiling Identifies
CRISPLD2</span> as a <span>Glucocorticoid Responsive Gene</span> that
<span>Modulates Cytokine Function</span> in <span>Airway Smooth Muscle
Cells</span>. PLOS ONE 2014;9:e99625. <a href="https://doi.org/10.1371/journal.pone.0099625" class="external-link">https://doi.org/10.1371/journal.pone.0099625</a>.</div>
</div>
<div id="ref-loveModeratedEstimationFold2014" class="csl-entry">
<div class="csl-left-margin">[2] </div>
<div class="csl-right-inline">Love MI, Huber W, Anders S. Moderated
estimation of fold change and dispersion for <span class="nocase">RNA-seq</span> data with <span>DESeq2</span>. Genome
Biology 2014;15:550. <a href="https://doi.org/10.1186/s13059-014-0550-8" class="external-link">https://doi.org/10.1186/s13059-014-0550-8</a>.</div>
</div>
<div id="ref-korotkevichFastGeneSet2016" class="csl-entry">
<div class="csl-left-margin">[3] </div>
<div class="csl-right-inline">Korotkevich G, Sukhov V, Budin N, Shpak B,
Artyomov MN, Sergushichev A. Fast gene set enrichment analysis.
<span>Bioinformatics</span>; 2016. <a href="https://doi.org/10.1101/060012" class="external-link">https://doi.org/10.1101/060012</a>.</div>
</div>
<div id="ref-subramanianGeneSetEnrichment2005" class="csl-entry">
<div class="csl-left-margin">[4] </div>
<div class="csl-right-inline">Subramanian A, Tamayo P, Mootha VK, Mukherjee
S, Ebert BL, Gillette MA, et al. Gene set enrichment analysis:
<span>A</span> knowledge-based approach for interpreting genome-wide
expression profiles. Proc Natl Acad Sci USA 2005;102:15545–50. <a href="https://doi.org/10.1073/pnas.0506580102" class="external-link">https://doi.org/10.1073/pnas.0506580102</a>.</div>
</div>
<div id="ref-liberzonMolecularSignaturesDatabase2011" class="csl-entry">
<div class="csl-left-margin">[5] </div>
<div class="csl-right-inline">Liberzon A, Subramanian A, Pinchback R,
Thorvaldsdóttir H, Tamayo P, Mesirov JP. Molecular signatures database
(<span>MSigDB</span>) 3.0. Bioinformatics 2011;27:1739–40. <a href="https://doi.org/10.1093/bioinformatics/btr260" class="external-link">https://doi.org/10.1093/bioinformatics/btr260</a>.</div>
</div>
<div id="ref-dolgalevMsigdbrMSigDBGene2022" class="csl-entry">
<div class="csl-left-margin">[6] </div>
<div class="csl-right-inline">Dolgalev I. <a href="https://cran.r-project.org/web/packages/msigdbr/index.html" class="external-link">Msigdbr:
<span>MSigDB Gene Sets</span> for <span>Multiple Organisms</span> in a
<span>Tidy Data Format</span></a>. 2022.</div>
</div>
<div id="ref-liberzonMolecularSignaturesDatabase2015" class="csl-entry">
<div class="csl-left-margin">[7] </div>
<div class="csl-right-inline">Liberzon A, Birger C, Thorvaldsdóttir H, Ghandi
M, Mesirov JP, Tamayo P. The <span>Molecular Signatures Database
Hallmark Gene Set Collection</span>. Cell Systems 2015;1:417–25. <a href="https://doi.org/10.1016/j.cels.2015.12.004" class="external-link">https://doi.org/10.1016/j.cels.2015.12.004</a>.</div>
</div>
<div id="ref-haunerGlucocorticoidsInsulinPromote1987" class="csl-entry">
<div class="csl-left-margin">[8] </div>
<div class="csl-right-inline">HAUNER H, SCHMID P, PFEIFFER EF.
Glucocorticoids and <span>Insulin Promote</span> the
<span>Differentiation</span> of <span>Human Adipocyte Precursor
Cells</span> into <span>Fat Cells</span>. The Journal of Clinical
Endocrinology &amp; Metabolism 1987;64:832–5. <a href="https://doi.org/10.1210/jcem-64-4-832" class="external-link">https://doi.org/10.1210/jcem-64-4-832</a>.</div>
</div>
<div id="ref-pantojaGlucocorticoidSignalingDefines2008" class="csl-entry">
<div class="csl-left-margin">[9] </div>
<div class="csl-right-inline">Pantoja C, Huff JT, Yamamoto KR. Glucocorticoid
<span>Signaling Defines</span> a <span>Novel Commitment State</span>
during <span>Adipogenesis In Vitro</span>. MBoC 2008;19:4032–41. <a href="https://doi.org/10.1091/mbc.e08-04-0420" class="external-link">https://doi.org/10.1091/mbc.e08-04-0420</a>.</div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Yi Jou (Ruby) Liao.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
